glean-core/src/metrics/labeled.rs

```diff
@@ -24,6 +28,46 @@ pub type LabeledBoolean = LabeledMetric<BooleanMetric>;
 /// A labeled string.
 pub type LabeledString = LabeledMetric<StringMetric>;
 
+/// A labeled custom_distribution.
+pub type LabeledCustomDistribution = LabeledMetric<CustomDistributionMetric>;
+
+/// A labeled memory_distribution.
+pub type LabeledMemoryDistribution = LabeledMetric<MemoryDistributionMetric>;
+
+/// A labeled timing_distribution.
+pub type LabeledTimingDistribution = LabeledMetric<TimingDistributionMetric>;
+
+/// The metric data needed to construct inner submetrics.
+///
+/// Different Labeled metrics require different amounts and kinds of information to
+/// be constructed.
+pub enum LabeledMetricData {
+    /// The common case: just a CMD.
+    #[allow(missing_docs)]
```

Comment:
I guess not documenting the fields here is fine.
You could move this above the `enum LabeledMetricData` and save the annotations per variant.

Reply:
Eh, I like it this way. Forces the next person to decide whether or not they want to document.